starlette/datastructures.py

```diff
@@ -604,6 +604,15 @@ def __delitem__(self, key: str) -> None:
         for idx in reversed(pop_indexes):
             del self._list[idx]
 
+    def __ior__(self, other: dict) -> "MutableHeaders":
+        self.update(other)
+        return self
+
+    def __or__(self, other: dict) -> "MutableHeaders":
+        new = self.mutablecopy()
+        new.update(other)
+        return new
```

Comment:
I guess we'd want to make sure it's a mapping before performing the `__or__` operation.

```suggestion
    def __or__(self, other: object) -> "MutableHeaders":
        if not isinstance(other, Mapping):
            return NotImplemented
        new = self.mutablecopy()
        new.update(other)
        return new
```